bleak/exc.py

```diff
@@ -9,6 +10,27 @@ class BleakError(Exception):
     pass
 
 
+class BluetoothPermissionErrorReason(enum.Enum):
+    # The bluetooth access was denied explicitly by the user.
+    Denied = enum.auto()
+
+    # The bluetooth access is restricted, possibly due to active restrictions such as parental controls.
+    Restricted = enum.auto()
```

Comment:
The difference between "denied" and "restricted" isn't very clear without reading the docs. The [analogous Windows API](https://learn.microsoft.com/en-us/uwp/api/windows.devices.radios.radioaccessstatus?view=winrt-26100) uses "denied by user" and "denied by system" which seems a bit more clear to me. So I would prefer that naming. Android just has [PERMISSION_DENIED](https://developer.android.com/reference/android/content/pm/PackageManager#PERMISSION_DENIED), but I would consider mapping that to "denied by user".